### ISSUE TITLE
[CI] Z3 version too old for use w/circt-lec

### ISSUE THREAD

**Issue by dtzSiFive** [llvm/circt#5827]
CI builds have this warning:

```
  Cannot build circt-lec with outdated Z3 version 4.8.7, requires 4.8.11.
```

(example: https://github.com/llvm/circt/actions/runs/5825770350/job/15798151008 )

That's awfully close, can we bump/provide a suitable Z3 so this can be built + tested as part of those flows?

Not sure if this is a GH runner environment or docker container thing, but flagging in case it's something we can get fixed without too much trouble :+1: .

**Comment by teqdruid** [comment]
It's a container thing. Do we have any normal tests which test the lec functionality? I only see tests in the integration_tests directory. The integration test image does have a newer version of z3 so circt-lec is compiled (and thus is checked by the short integration tests in each PR).

**Comment by dtzSiFive** [comment]
Oh okay! Thanks, that sounds like all is well!
